src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java

```diff
@@ -906,4 +909,26 @@ public void testPersistentTokenCacheAccessAspect() throws SQLException {
         persistentTokenAspect.beforeCacheAccess(tokenCacheAccessContext);
         PersistentTokenCacheAccessAspect.clearUserTokenCache();
     }
+
+    /**
+     * test connection properties
+     * 
+     * @throws SQLException
+     */
+    @Test
+    public void testServerCert() throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        ds.setURL(connectionString);
+        ds.setServerCertificate("badCert");
+        ds.setEncrypt(Constants.STRICT);
+        ds.setTrustServerCertificate(false);
+        try (Connection con = ds.getConnection()) {
+            fail(TestResource.getResource("R_expectedFailPassed"));
+        } catch (SQLException e) {
+            // TODO: servers which do not support TDSS will return SSL failed error, test should be updated once server

```

Comment:
Hmm, I was going to ask about more tests eg. maybe one test that tests the connection string, but without servers that support TDSS, it doesn't seem possible?